plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -174,61 +172,65 @@ public CompletableFuture<ConnectorSplitBatch> getNextBatch(ConnectorPartitionHan
             return completedFuture(NO_MORE_SPLITS_BATCH);
         }
 
-        Iterator<FileScanTask> fileScanTasks = Iterators.limit(fileScanTaskIterator, maxSize);
```

Comment:
in "Do not count filtered splits against max split count in Iceberg" commit

please also add the Why? for the change.

- pros: it returns p to `maxSize` splits per invocation
- cons: it can do a lot of work (eg processing millions of splits) before returning anything

... i am actually not convinced we should do the change.
if we do, or if we do not, we need explanation as a code comment too.

Reply:
> cons: it can do a lot of work (eg processing millions of splits) before returning anything

I don't think about that, I can walk that commit back. I didn't actually compare if it was any better